test/lib/CustomRulePathManager.test.ts

```diff
@@ -0,0 +1,176 @@
+    afterEach(() => { sandbox.restore(); });
+
+    // File exists, but content has been somehow cleared
+    const emptyFile = '';
+
+    // One entry for apex, two for java
+    const populatedFile = '{"pmd": {"apex": ["/some/user/path/customRule.jar"],"java": ["/abc/def/ghi","/home/lib/jars"]}}';
+
+    describe('Rule path entries creation', () => {
+        it('should read CustomPaths.json to get Rule Path Entries', async () => {
+            // Setup stub
+            let stub = Sinon.createStubInstance(FileIOHandler);
+            stub.readFile.resolves(populatedFile);
+            const manager = new CustomRulePathManager(stub);
+
+            // Execute test
+            const rulePathMap = await manager.getRulePathEntries(ENGINE.PMD);
+            rulePathMap.forEach((value, key) => {
+                console.log(`${key}: ${Array.from(value)}`);
+            });
+
+            // Validate run
+            expect(stub.readFile.calledOnce).to.be.true;
+            expect(rulePathMap).to.be.lengthOf(2);
+
+            //Validate each entry
+            expect(rulePathMap).has.keys('apex', 'java');
+            const apexPaths = rulePathMap.get('apex');
+            expect(apexPaths).to.be.lengthOf(1);
+            expect(apexPaths).deep.contains('/some/user/path/customRule.jar');
+
+            const javaPaths = rulePathMap.get('java');
+            expect(javaPaths).to.be.lengthOf(2);
+            expect(javaPaths).deep.contains('/abc/def/ghi');
+            expect(javaPaths).deep.contains('/home/lib/jars');
+        });
+
+        it('should handle empty Rule Path file gracefully', async () => {
+            // Setup stub
+            let stub = Sinon.createStubInstance(FileIOHandler);
+            stub.readFile.resolves(emptyFile);
+            const manager = new CustomRulePathManager(stub);
+
+            // Execute test
+            const rulePathMap = await manager.getRulePathEntries(ENGINE.PMD);
+
+            // Validate
+            expect(rulePathMap).is.not.null;
+            expect(rulePathMap).to.be.lengthOf(0);
+
+        });
+
+        // it('should handle non-existent CustomPath.json file gracefully', async () => {
+        //     // Setup stub
+        //     let stub = Sinon.createStubInstance(FileOperations);
+        //     const dummyError = new Error('Test error to indicate file does not exist');
+
+        //     // TODO: how do I set code as ENOENT?
+        //     stub.readRulePathFile.throws(dummyError);
+        //     const manager = new CustomRulePathManager(stub);
+
+        //     // Execute test and validate
+
+        //     await manager.getRulePathEntries(ENGINE.PMD);
+        //     expect(await manager.getRulePathEntries.bind(ENGINE.PMD)).throws(messages.getMessage('errors.readCustomRulePathFileFailed'));
+
+        //     // Validate
+
+
+        // });
+
+        it('should initialize only once', async () => {
+            // Setup stub
+            let stub = Sinon.createStubInstance(FileIOHandler);
```

Comment:
If you use `Sinon.stub(FileIOHandler.prototype, 'readFile').callsFake(async () => {return populatedFile});`, you won't need to pass a `FileIOHandler` into the constructor for `CustomRulePathManager` anymore. I can show you that if you like.

Reply:
Ah, good find. Let me try it out.

Reply:
Found an even more concise way to stub:
`readStub = Sinon.stub(FileHandler.prototype, 'readFile').resolves(populatedFile);`